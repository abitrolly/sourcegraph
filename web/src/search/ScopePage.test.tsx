import React from 'react'
import { MemoryRouter } from 'react-router'
import { ScopePage } from './ScopePage'
import * as H from 'history'
import { SearchPatternType } from '../../../shared/src/graphql/schema'
import { mount } from 'enzyme'

jest.mock('./input/QueryInput', () => ({ QueryInput: 'QueryInput' }))
jest.mock('./input/SearchButton', () => ({ SearchButton: 'SearchButton' }))

const history = H.createMemoryHistory()
const location = H.createLocation('/')

describe('ScopePage', () => {
    test('renders', () =>
        expect(
            mount(
                <MemoryRouter>
                    <ScopePage
                        match={{ params: { id: 'my-scope-id' }, isExact: true, path: '/', url: '/' }}
                        settingsCascade={{
                            final: {
                                'search.scopes': [
                                    { id: 'my-scope-id', description: 'my description', value: 'my-scope-value' },
                                ],
                            },
                            subjects: [],
                        }}
                        authenticatedUser={null}
                        caseSensitive={false}
                        onNavbarQueryChange={() => undefined}
                        patternType={SearchPatternType.literal}
                        setCaseSensitivity={() => undefined}
                        setPatternType={() => undefined}
                        history={history}
                        location={location}
                        copyQueryButton={false}
                        versionContext={undefined}
                    />
                </MemoryRouter>
            )
        ).toMatchSnapshot())
})
