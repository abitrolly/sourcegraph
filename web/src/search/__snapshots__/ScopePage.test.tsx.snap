// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScopePage renders 1`] = `
<MemoryRouter>
  <Router
    history="[History]"
  >
    <ScopePage
      authenticatedUser={null}
      caseSensitive={false}
      copyQueryButton={false}
      history="[History]"
      location="[Location path=/]"
      match={
        Object {
          "isExact": true,
          "params": Object {
            "id": "my-scope-id",
          },
          "path": "/",
          "url": "/",
        }
      }
      onNavbarQueryChange={[Function]}
      patternType="literal"
      setCaseSensitivity={[Function]}
      setPatternType={[Function]}
      settingsCascade={
        Object {
          "final": Object {
            "search.scopes": Array [
              Object {
                "description": "my description",
                "id": "my-scope-id",
                "value": "my-scope-value",
              },
            ],
          },
          "subjects": Array [],
        }
      }
    >
      <div
        className="container mt-3"
      >
        <PageTitle />
        <header>
          <h1 />
          <Markdown
            dangerousInnerHTML="<p>my description</p>
"
            history="[History]"
          >
            <div
              className="markdown"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<p>my description</p>
",
                }
              }
              onClick={[Function]}
            />
          </Markdown>
        </header>
        <section
          className="mb-5"
        >
          <Form
            className="d-flex"
            onSubmit={[Function]}
          >
            <form
              className="d-flex "
              onInvalid={[Function]}
              onSubmit={[Function]}
            >
              <QueryInput
                authenticatedUser={null}
                autoFocus={true}
                caseSensitive={false}
                copyQueryButton={false}
                history="[History]"
                location="[Location path=/]"
                match={
                  Object {
                    "isExact": true,
                    "params": Object {
                      "id": "my-scope-id",
                    },
                    "path": "/",
                    "url": "/",
                  }
                }
                onChange={[Function]}
                patternType="literal"
                placeholder="Search..."
                prependQueryForSuggestions="my-scope-value"
                setCaseSensitivity={[Function]}
                setPatternType={[Function]}
                settingsCascade={
                  Object {
                    "final": Object {
                      "search.scopes": Array [
                        Object {
                          "description": "my description",
                          "id": "my-scope-id",
                          "value": "my-scope-value",
                        },
                      ],
                    },
                    "subjects": Array [],
                  }
                }
                value={
                  Object {
                    "cursorPosition": 0,
                    "query": "",
                  }
                }
              />
              <SearchButton />
            </form>
          </Form>
          <div
            className="d-flex align-items-center m-1 text-muted"
          >
            <span
              className="mr-1"
            >
              Scope:
            </span>
            <code
              className="border rounded p-1"
            >
              my-scope-value
            </code>
          </div>
        </section>
        <div
          className="alert alert-warning"
        >
          No repositories in scope.
        </div>
      </div>
    </ScopePage>
  </Router>
</MemoryRouter>
`;
